### DISCUSSION TITLE
how can I modify the menu on the tab freely?

### DISCUSSION THREAD

**Discussion by bezierC0** [fluentribbon/Fluent.Ribbon#1205] [Q&A] [answered]
I have three questions about this menu

1. Can I change the menu language in the settings?
2. Can I hide this menu?
3. Can I change the text of the menu items?

![Snipaste_2024-06-17_11-25-37](https://github.com/fluentribbon/Fluent.Ribbon/assets/95664753/63b32245-9b47-4570-9d85-b1919e4cbabd)

**Comment by batzen** [comment]
1. The language used there is the current UI culture and it can be changed by using `RibbonLocalization.Current.Culture`
2. You can hide it by setting `IsToolBarVisible` on `Ribbon` or directly on `RibbonTabControl`
3. The only way to change the text of those menu items is by providing your own localization and supplying things like `ShowRibbon`, `ExpandRibbon` etc.

**Reply by bezierC0** [reply]
@batzen 
Thank you for your reply.
1. How to use RibbonLocalization.Current.Culture to change the language? Can you write a specific sample code?
2. Add IsToolBarVisible to Fluent:Ribbon and set it to false, but it still displays. The specific code is as follows.
```
        <Fluent:Ribbon IsToolBarVisible="False" >
        </Fluent:Ribbon>
```
3. How to provide my own localization and supplying things? Do I need to rewrite the language class of Fluent.Localization.Languages? Can you provide a specific code sample?

**Reply by batzen** [reply, marked as answer]
1. `RibbonLocalization.Current.Culture = new System.Globalization.CultureInfo("de");`
2. Sorry. You have to set `IsDisplayOptionsButtonVisible` to hide that button.
3. You can either use a specific existing implementation like `RibbonLocalization.Current.Localization = new German();` or you could create a new class, inherit from `RibbonLocalizationBase` and use that one then.
